processing/src/main/java/org/apache/druid/query/filter/vector/FalseVectorMatcher.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.filter.vector;
+
+import org.apache.druid.segment.vector.VectorSizeInspector;
+
+public class FalseVectorMatcher implements VectorValueMatcher
+{
+  private final VectorSizeInspector vectorSizeInspector;
+
+  public FalseVectorMatcher(VectorSizeInspector vectorSizeInspector)
+  {
+    this.vectorSizeInspector = vectorSizeInspector;
+  }
+
+  @Override
+  public ReadableVectorMatch match(ReadableVectorMatch mask)
+  {
+    return VectorMatch.allFalse();
+  }
+
+  @Override
+  public int getMaxVectorSize()
+  {
+    return vectorSizeInspector.getMaxVectorSize();
+  }
+
+  @Override
+  public int getCurrentVectorSize()
+  {
+    return vectorSizeInspector.getCurrentVectorSize();
+  }
```

Comment:
Should the max size / current size of these be `1` since `VectorMatch.allFalse()` is an empty array?

Not really sure how the vector sizes are used

Reply:
The max vector size is the maximum number of values in a column you can read per iteration. The current vector size can be different depending on the remaining number of values to read.

The max and current vector sizes are irrelevant to the size of backing array in `VectorMatcher`, although the array usually has the same size with the max vector size. The allFalse vectorMatch has an empty array because it always returns false no matter what value you evaluate in a vector.

Reply:
ah cool. Thanks for the explanation!